packages/flutter/lib/src/services/binary_messenger.dart

```diff
@@ -0,0 +1,153 @@
+// Copyright 2017 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:async';
+import 'dart:typed_data';
+import 'dart:ui' as ui;
+
+import 'package:flutter/foundation.dart';
+
+typedef _MessageHandler = Future<ByteData> Function(ByteData message);
+
+/// A messenger which sends binary data across the Flutter platform barrier.
+///
+/// This class also registers handlers for incoming messages.
+abstract class BinaryMessenger {
+  /// A const constructor to allow subclasses to be const.
+  const BinaryMessenger();
+
+  /// Calls the handler registered for the given channel.
+  ///
+  /// Typically called by [ServicesBinding] to handle platform messages received
+  /// from [Window.onPlatformMessage].
+  ///
+  /// To register a handler for a given message channel, see [setMessageHandler].
+  Future<void> handlePlatformMessage(String channel, ByteData data, ui.PlatformMessageResponseCallback callback);
+
+  /// Send a binary message to the platform plugins on the given channel.
+  ///
+  /// Returns a [Future] which completes to the received response, undecoded,
+  /// in binary form.
+  Future<ByteData> send(String channel, ByteData message);
+
+  /// Set a callback for receiving messages from the platform plugins on the
+  /// given channel, without decoding them.
+  ///
+  /// The given callback will replace the currently registered callback for that
+  /// channel, if any. To remove the handler, pass null as the [handler]
+  /// argument.
+  ///
+  /// The handler's return value, if non-null, is sent as a response, unencoded.
+  void setMessageHandler(String channel, Future<ByteData> handler(ByteData message));
+
+  /// Set a mock callback for intercepting messages from the [send] method on
+  /// this class, on the given channel, without decoding them.
+  ///
+  /// The given callback will replace the currently registered mock callback for
+  /// that channel, if any. To remove the mock handler, pass null as the
+  /// [handler] argument.
+  ///
+  /// The handler's return value, if non-null, is used as a response, unencoded.
+  ///
+  /// This is intended for testing. Messages intercepted in this manner are not
+  /// sent to platform plugins.
+  void setMockMessageHandler(String channel, Future<ByteData> handler(ByteData message));
+}
+
+/// The default implementation of [BinaryMessenger].
+///
+/// This messenger sends messages from the app-side to the platform-side and
+/// dispatches incoming messages from the platform-side to the appropriate
+/// handler.
+class _DefaultBinaryMessenger extends BinaryMessenger {
+  const _DefaultBinaryMessenger._();
+
+  // Handlers for incoming messages from platform plugins.
+  // Note: This is static so that this class can have a const constructor.
```

Comment:
here are below: no need to say "Note:". See https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#avoid-empty-prose

Reply:
Done.